packages/inspection-report/src/components/common/DamageManipulator.js

```diff
@@ -0,0 +1,160 @@
+/* eslint-disable react/no-unescaped-entities */
+import React, { useState } from 'react';
+import { StyleSheet, Text, TouchableOpacity, View } from 'react-native';
+import Slider from '@react-native-community/slider';
+import PropTypes from 'prop-types';
+import { useTranslation } from 'react-i18next';
+import { MaterialIcons } from '@expo/vector-icons';
+import moderate from './../../assets/moderate.svg';
+
+import SwitchButton from './SwitchButton';
+
+const styles = StyleSheet.create({
+  container: {
+    display: 'flex',
+  },
+  severityButtonWrapper: {
+    alignItems: 'center',
+    borderColor: '#5d5e67',
+    borderRadius: 8,
+    borderWidth: 2,
+    display: 'flex',
+    flexDirection: 'row',
+    marginRight: 8,
+    padding: 15,
+  },
+  severityText: {
+    fontSize: 16,
+    marginLeft: 10,
+  },
+  damageWrapper: {
+    alignItems: 'center',
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+  },
+  title: {
+    fontSize: 14,
+    lineHeight: 20,
+    opacity: '0.72',
+  },
+  button: {
+    borderColor: '#5d5e67',
+    borderRadius: 28,
+    borderWidth: 2,
+    marginTop: 10,
+    padding: 10,
+  },
+  text: {
+    fontSize: 14,
+    textAlign: 'center',
+  },
+  disabled: {
+    opacity: 0.5
+  },
+  moderateIconContainer: {
+    alignItems: 'flex-end',
+    backgroundColor: '#d4e157',
+    borderRadius: 24,
+    display: 'flex',
+    height: 24,
+    justifyContent: 'center',
+    paddingHorizontal: 3,
+    paddingVertical: 5,
+    width: 24,
+  },
+  moderateIcon: {
+    backgroundColor: '#000',
+    border: 5,
+    borderBottomRightRadius: 12,
+    borderTopRightRadius: 12,
+    height: 16,
+    width: 8,
+  },
+});
+
+function DamageManipulator({ damageMode, displayMode, onConfirm, damage }) {
+  const { t } = useTranslation();
+
+  const [isEnabled, setIsEnabled] = useState(false);
+  const toggleSwitch = () => setIsEnabled((previousState) => !previousState);
+  const [severity, setSeverity] = useState('');
+  const [pricing, setPricing] = useState(0);
```

Comment:
No this is not how you are supposed to implement this. If you take a look at the docs you should have an edited damage as a state and use the `damage` prop as the initial value. Please do something like :

```javascript
// Declare this OUTSIDE of the component
const initialDamage = {
  severity: { severity: 'low' },
  pricing: { pricing: 0 },
  all: { severity: 'low', pricing: 0 },
};

// Inside the component
const [editedDamage, setEditedDamage] = useState(damage);
const initialDamageValue = useMemo(() => {
  if (damageMode === 'severity') {
    return 
}, []);
const toggleSwitch = useCallback(
  () => setEditedDamage((dmg) => (dmg ? null : initialDamage[damageMode])),
  [damageMode],
);
```

And then the value of the slider should be linked to `editedDamage.pricing` and the selected severity should be linked to `editedDamage.severity`. Also, to know wether or not the checkbox is checked or not, you just have to check if `editedDamage` is defined

Reply:
ah okay. Thanks

Reply:
Resolved